From 394c17bd424670bc8e46397f8d925f124549d1ab Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Sun, 4 Apr 2021 18:31:07 +0300
Subject: [PATCH 1414/1921] arm64: dts: add Marvell PCIe EP platforms

This patch adds CN913x platforms configured to work in PCIe EP
mode (configuration "C")

Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Change-Id: I55850b2cd4d8ba56b69848cc3008bb254217287f
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/49166
Tested-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Ofer Heifetz <oferh@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/arm64/boot/dts/marvell/Makefile         |  4 +
 arch/arm64/boot/dts/marvell/cn9130-crb-C.dts | 14 ++++
 arch/arm64/boot/dts/marvell/cn9130-db-C.dts  | 24 ++++++
 arch/arm64/boot/dts/marvell/cn9131-db-C.dts  | 22 +++++
 arch/arm64/boot/dts/marvell/cn9132-db-C.dts  | 21 +++++
 arch/arm64/boot/dts/marvell/cn913x-ep.dtsi   | 85 ++++++++++++++++++++
 6 files changed, 170 insertions(+)
 create mode 100644 arch/arm64/boot/dts/marvell/cn9130-crb-C.dts
 create mode 100644 arch/arm64/boot/dts/marvell/cn9130-db-C.dts
 create mode 100644 arch/arm64/boot/dts/marvell/cn9131-db-C.dts
 create mode 100644 arch/arm64/boot/dts/marvell/cn9132-db-C.dts
 create mode 100644 arch/arm64/boot/dts/marvell/cn913x-ep.dtsi

diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 86549df7accc..75be3581b036 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -15,9 +15,13 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-mcbin-singleshot.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8080-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9130-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9130-db-B.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += cn9130-db-C.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9131-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9131-db-B.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += cn9131-db-C.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9132-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9132-db-B.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += cn9132-db-C.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9130-crb.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9130-crb-B.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += cn9130-crb-C.dtb
diff --git a/arch/arm64/boot/dts/marvell/cn9130-crb-C.dts b/arch/arm64/boot/dts/marvell/cn9130-crb-C.dts
new file mode 100644
index 000000000000..3c267f6cf888
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/cn9130-crb-C.dts
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2021 Marvell International Ltd.
+ *
+ * Device tree for the CN9130-CRB-C board.
+ */
+
+#include "cn9130-crb.dtsi"
+#include "cn913x-ep.dtsi"
+
+/ {
+	model = "Marvell Armada CN9130-CRB-C";
+};
+
diff --git a/arch/arm64/boot/dts/marvell/cn9130-db-C.dts b/arch/arm64/boot/dts/marvell/cn9130-db-C.dts
new file mode 100644
index 000000000000..34dff0f7aeff
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/cn9130-db-C.dts
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2021 Marvell International Ltd.
+ *
+ * Device tree for the CN9130-DB-C board.
+ */
+
+#include "cn9130.dtsi"
+#include "cn913x-ep.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	model = "Marvell Armada CN9130-DB setup C";
+};
+
+/* Setup C has SPI1 flash as a boot device, while setup B uses NAND flash.
+ * Since CP0 SPI1 and CP0 NAND are sharing some pins, they cannot be activated
+ * simultaneously. When SPI controller is enabled, NAND should be disabled.
+ */
+&cp0_spi1 {
+	status = "okay";
+};
+
diff --git a/arch/arm64/boot/dts/marvell/cn9131-db-C.dts b/arch/arm64/boot/dts/marvell/cn9131-db-C.dts
new file mode 100644
index 000000000000..4d0fb24e51fa
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/cn9131-db-C.dts
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2021 Marvell International Ltd.
+ *
+ * Device tree for the CN9131-DB-C board.
+ */
+
+#include "cn9131-db.dtsi"
+#include "cn913x-ep.dtsi"
+
+/ {
+	model = "Marvell Armada CN9131-DB setup C";
+};
+
+/* Setup A has SPI1 flash as a boot device, while setup B uses NAND flash.
+ * Since CP0 SPI1 and CP0 NAND are sharing some pins, they cannot be activated
+ * simultaneously. When SPI controller is enabled, NAND should be disabled.
+ */
+&cp0_spi1 {
+	status = "okay";
+};
+
diff --git a/arch/arm64/boot/dts/marvell/cn9132-db-C.dts b/arch/arm64/boot/dts/marvell/cn9132-db-C.dts
new file mode 100644
index 000000000000..85dccf982c01
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/cn9132-db-C.dts
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2021 Marvell International Ltd.
+ *
+ * Device tree for the CN9132-DB C board.
+ */
+
+#include "cn9132-db.dtsi"
+#include "cn913x-ep.dtsi"
+
+/ {
+	model = "Marvell Armada CN9132-DB setup C";
+};
+
+/* Setup A has SPI1 flash as a boot device, while setup B uses NAND flash.
+ * Since CP0 SPI1 and CP0 NAND are sharing some pins, they cannot be activated
+ * simultaneously. When SPI controller is enabled, NAND should be disabled.
+ */
+&cp0_spi1 {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/marvell/cn913x-ep.dtsi b/arch/arm64/boot/dts/marvell/cn913x-ep.dtsi
new file mode 100644
index 000000000000..1f5cd2060e11
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/cn913x-ep.dtsi
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2021 Marvell International Ltd.
+ *
+ * Device tree include for the CN913x End Points.
+ */
+
+/ {
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		/* The PCI console memory must be reserved */
+		console_reserved: pci-console-nexus@3f000000 {
+			compatible = "marvell,pci-console-nexus-memory";
+			reg = <0 0x3f000000 0 0x1000000>;
+			no-map;
+		};
+	};
+
+	cp0 {
+		/delete-node/ pcie@600000;
+		/delete-node/ pcie@620000;
+		/delete-node/ pcie@640000;
+	};
+};
+
+&smmu {
+	/* Should only be enabled on RC side */
+	status = "disabled";
+};
+
+&dma_xor0 {
+	status = "okay";
+};
+
+&dma_xor1 {
+	status = "okay";
+};
+
+&xor0 {
+	status = "disabled";
+};
+
+&xor1 {
+	status = "disabled";
+};
+
+&uio_xor0 {
+	status = "disabled";
+};
+
+&uio_xor1 {
+	status = "disabled";
+};
+
+&cp0_pci_ep_uio {
+	status = "okay";
+};
+
+&cp0_pcie_ep {
+	status = "okay";
+};
+
+&cp0_comphy0 {
+	phy-skip-config;
+};
+
+&cp0_comphy1 {
+	phy-skip-config;
+};
+
+&cp0_comphy2 {
+	phy-skip-config;
+};
+
+&cp0_comphy3 {
+	phy-skip-config;
+};
+
+&cp0_armada_ep {
+	status = "okay";
+};
+
-- 
2.31.1

